File: app/src/main/java/com/mobarak/todo/ui/statistics/StatisticsFragment.java
package com.mobarak.todo.ui.statistics;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.mobarak.todo.R;
import com.mobarak.todo.ui.base.ViewModelFactory;
import com.mobarak.todo.data.AppRepositoryImpl;
import com.mobarak.todo.databinding.StatisticsFragBinding;
import com.mobarak.todo.utility.ViewUtil;

public class StatisticsFragment extends Fragment {

    private StatisticsViewModel viewModel;
    private StatisticsFragBinding viewDataBinding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.statistics_frag, container, false);
        viewDataBinding = StatisticsFragBinding.bind(root);
        return root;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        viewModel = new ViewModelProvider(requireActivity(),
                new ViewModelFactory(getActivity(), AppRepositoryImpl.getInstance(), this)
        ).get(StatisticsViewModel.class);
        viewDataBinding.setLifecycleOwner(this);
        viewDataBinding.setViewmodel(viewModel);
        ViewUtil.setupRefreshLayout(getActivity(), viewDataBinding.refreshLayout, null);
        viewModel.refresh();

    }
}

File: app/src/main/java/com/mobarak/todo/ui/statistics/StatisticsFragment.kt
package com.mobarak.todo.ui.statistics

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import com.mobarak.todo.R
import com.mobarak.todo.databinding.StatisticsFragBinding
import com.mobarak.todo.utility.getViewModelFactory
import com.mobarak.todo.utility.setupRefreshLayout

class StatisticsFragment : Fragment() {
    private val viewModel by viewModels<StatisticsViewModel> { getViewModelFactory() }
    private lateinit var viewDataBinding: StatisticsFragBinding
    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?, savedInstanceState: Bundle?): View {
        viewDataBinding = DataBindingUtil.inflate(
                inflater, R.layout.statistics_frag, container,
                false
        )
        return viewDataBinding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewDataBinding.viewmodel = viewModel
        viewDataBinding.lifecycleOwner = this.viewLifecycleOwner
        this.setupRefreshLayout(viewDataBinding.refreshLayout)
    }
}